Bolts/Common/BFTask.m

```diff
@@ -437,26 +474,26 @@ - (void)waitUntilFinished {
 
 - (NSString *)description {
     // Acquire the data from the locked properties
-    BOOL completed;
-    BOOL cancelled;
-    BOOL faulted;
-    NSString *resultDescription = nil;
-
-    @synchronized(self.lock) {
-        completed = self.completed;
-        cancelled = self.cancelled;
-        faulted = self.faulted;
-        resultDescription = completed ? [NSString stringWithFormat:@" result = %@", self.result] : @"";
-    }
+    __block BOOL isCompleted;
+    __block BOOL isCancelled;
+    __block BOOL isFaulted;
+    __block NSString *resultDescription = nil;
+
+    dispatch_barrier_sync(_synchronizationQueue, ^{
+        isCompleted = _state != pending;
```

Comment:
Just take the state, result pair out, don't do any additional work inside the lock. Acquire the lock for as little time as is safely possible.